src/flamegraph/color.rs

```diff
@@ -0,0 +1,272 @@
+        }
+    };
+
+    for character in name.bytes() {
+        let i = (character % modulo) as f32;
+        vector += (i / ((modulo - 1) as f32)) * weight;
+        modulo += 1;
+        max += weight;
+        weight *= 0.70;
+
+        if modulo > 12 {
+            break
+        }
+    }
+
+    (1f32 - vector / max)
+}
+
+fn handle_java_palette(s: &str) -> Palette {
+    if s.ends_with("]") {
+        if let Some(ai) = s.rfind("_[") {
+            if s[ai..].len() == 4 {
+                let suffix_char = &s[ai+2..ai+3];
+                if suffix_char.starts_with("k") { return Palette::Orange }
+                if suffix_char.starts_with("i") { return Palette::Aqua }
+                if suffix_char.starts_with("j") { return Palette::Green }
+            }
+        }
+    }
+
+    let java_prefix = if s.starts_with("L") { &s[1..] } else { s };
+
+    if java_prefix.starts_with("java/") ||
+        java_prefix.starts_with("org/") ||
+        java_prefix.starts_with("com/") ||
+        java_prefix.starts_with("io/") ||
+        java_prefix.starts_with("sun/") {
+        Palette::Green
+    } else if s.contains("::") {
+        Palette::Yellow
+    } else {
+        Palette::Red
+    }
+}
+
+fn handle_perl_palette(s: &str) -> Palette {
+    if s.contains("::") {
+        Palette::Yellow
+    } else if s.contains("Perl") || s.contains(".pl") {
+        Palette::Green
+    } else if s.ends_with("_[k]") {
+        Palette::Orange
+    } else {
+        Palette::Red
+    }
+}
+
+fn handle_js_palette(s: &str) -> Palette {
+    if s.ends_with("_[j]") {
+        if s.contains("/") {
+            return Palette::Green
+        } else {
+            return Palette::Aqua
+        }
+    } else if s.contains("::") {
+        return Palette::Yellow
+    } else if let Some(ai) = s.find("/") {
+        if (&s[ai..]).contains(".js") {
+            return Palette::Green
+        }
+    }
+
+    if s.contains(":") {
+        Palette::Aqua
+    } else if s == " " {
+        Palette::Green
+    } else if s.contains("_[k]") {
+        Palette::Orange
+    } else {
+        Palette::Red
+    }
+}
+
+fn handle_wakeup_palette(_s: &str) -> Palette {
+    Palette::Aqua
+}
+
+fn handle_chain_palette(s: &str) -> Palette {
+    if s.contains("_[w]") {
+        Palette::Aqua
+    } else {
+        Palette::Blue
+    }
+}
+
+fn coefficients_for_palette(palette: &Palette, name: &str, v1: f32, v2: f32, v3: f32) ->
+    ((u8, u8, f32), (u8, u8, f32), (u8, u8, f32)) {
+    let real_palette = match palette {
+        Palette::Hot => return ((205, 50, v3), (0, 230, v1), (0, 55, v2)),
+        Palette::Mem => return ((0, 0, v3), (190, 50, v2), (0, 210, v1)),
+        Palette::Io => return ((80, 60, v1), (80, 60, v1), (190, 55, v2)),
+        Palette::Red => return ((200, 55, v1), (50, 80, v1), (50, 80, v1)),
+        Palette::Green => return ((50, 60, v1), (200, 55, v1), (50, 60, v1)),
+        Palette::Blue => return ((80, 60, v1), (80, 60, v1), (205, 60, v1)),
```

Comment:
I think the last component here should be `(205, 50, v1)`?

Reply:
[Oops](https://github.com/jonhoo/inferno/pull/32/commits/d5000681e16a220684e72429b32d5115588c483a)